src/server/src/main/java/io/cassandrareaper/service/SegmentRunner.java

```diff
@@ -295,6 +295,15 @@ private boolean runRepair() {
 
       LazyInitializer<Set<String>> busyHosts = new BusyHostsInitializer(cluster);
 
+      // If we're using a distributed storage, we need to synchronize with other Reaper instances
+      // So we don't start too many segments at the same time
+      if (!lockSegmentRunners()) {
+        LOG.debug(
+            "Not allowed to run the segment for now as another Reaper holds the lock for repair run {}. "
+            + "Will try again later",
+            segment.getRunId());
+        return false;
+      }
```

Comment:
i'm keen to minimise the need for this locking, as described in https://github.com/thelastpickle/cassandra-reaper/pull/601#commitcomment-31906618

by redo-ing and offering https://github.com/thelastpickle/cassandra-reaper/compare/sidecar-mode...mck/sidecar-mode

Reply:
work-in-progress here: https://github.com/thelastpickle/cassandra-reaper/pull/624

the idea is that PR#624 would get merged to master first.

Reply:
#624 has been merged to master.

off-line discussion agreed that side-car could add locking on top on the condition that it was optimistic: that is was only used when #624 was insufficient; and would be logged when required. This way over time the locking can be removed if it is proven to never be used.